CRM/Contribute/Form/AdditionalPayment.php

```diff
@@ -527,20 +490,28 @@ public function processCreditCard($submittedValues) {
         $result = $payment->doPayment($paymentParams);
       }
       catch (\Civi\Payment\Exception\PaymentProcessorException $e) {
-        //set the contribution mode.
-        $urlParams = "action=add&cid={$this->_contactId}&id={$this->_id}&component={$this->_component}";
-        if ($this->_mode) {
-          $urlParams .= "&mode={$this->_mode}";
-        }
-        CRM_Core_Error::displaySessionError($result);
-        CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/payment/add', $urlParams));
+        Civi::log()->error('Payment processor exception: ' . $e->getMessage());
+        $errorMessage = $e->getMessage();
       }
     }
 
-    if ($result) {
-      $this->_params = array_merge($this->_params, $result);
+    // If payment is not completed then bounce approrptiate status message
```

Comment:
isn't it more readable to keep all this in the catch block? I can't see anywhere else where $errorMessage is set.